### ISSUE TITLE
Possibility to fill the username too

### ISSUE THREAD

**Issue by Dschubba** [ttyridal/masterpassword-firefox#30]
Hi, 

it would be nice if the addon also fills the username that is stored by the addon. Is this possible?

**Comment by ttyridal** [comment]
Yeah, I've wanted this myself too. Largest problem is to find a good heuristic for locating the username field.  Passwords are easy, they have their own type.

**Comment by promeneur** [comment]
i suggest this:
make a fork of the FF add-on "KDE5 Wallet password integration" 
https://addons.mozilla.org/fr/firefox/addon/kde5-wallet-password-integrati/

you must only adapt it to masterpassword

there is also a FF add-on for gnome keyring
https://addons.mozilla.org/fr/firefox/addon/gnome-keyring-integration/

**Comment by ttyridal** [comment]
Released in 2.1.0 beta 6

**Comment by promeneur** [comment]
did you mean that we can inactivate "id,pwd recording & filling template" FF feature?

**Comment by ttyridal** [comment]
TLDR; If the behaviour of MP (described below) is good enough for you: Yes.

It works like this:
If the login form is according to common guidelines:
* username input should be of type text|email|num|tel
* It should have a name attribute containing any of user/name/email/login
* the next input is a type=password

MP will, if focused on such an input and if the input is empty, inject the username stored with the site settings in MP. Then advance to the password field and inject the password.

Many (most?) sites will adhere to this, with countless exceptions I presume.

If focused on an input[type=password], current behaviour (ie inject password) is invoked.

I've tried to strike a balance between "working on most sites" and preventing accidental data leaks by entering login details in random forms.

**Comment by promeneur** [comment]
ok i will try.

thanks for your job
